names/names.py

```diff
@@ -11,12 +67,16 @@
 f.close()
 
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
 
-end_time = time.time()
-print (f"{len(duplicates)} duplicates:\n\n{', '.join(duplicates)}\n\n")
-print (f"runtime: {end_time - start_time} seconds")
+name_dup = BinarySearchTree('new_name')
```

Comment:
Nice work utilizing the BST we made earlier in this week to take advantage of O(log(n)) lookup.  

In the future, consider using a hash-table or set for this type of problem.  Since they have O(1) lookup, this would change the overall runtime complexity from O(nlog(n)) to just O(n).